chia/wallet/puzzles/curry.clinc

```diff
@@ -0,0 +1,68 @@
+(
+  ;; The code below is used to calculate of the tree hash of a curried function
+  ;; without actually doing the curry, and using other optimization tricks
+  ;; like unrolling `sha256tree`.
+
+  (defconstant ONE 1)
+  (defconstant TWO 2)
+  (defconstant A_KW #a)
+  (defconstant Q_KW #q)
+  (defconstant C_KW #c)
+	(defconstant sha256_one 0x4bf5122f344554c53bde2ebb8cd2b7e3d1600ad631c385a5d7cce23c7785459a)
```

Comment:
So writing `(sha 1)` in multiple places would expand out to the constant. I figured since I can't keep the long constant out of the code, I may as well just include it once, as a `defconstant`.